modules/tb-all-in-one/main.tf

```diff
@@ -0,0 +1,53 @@
+resource "digitalocean_volume" "storage" {
+  region      = "nyc1"
+  name        = "storage-${var.name}"
+  size        = 10
+  description = "${var.name} Volume"
+}
+
+resource "digitalocean_droplet" "droplet-1" {
+  image = "ubuntu-16-04-x64"
+  name = "${var.name}-1"
+
+  region = "${var.region}"
+  size = "${var.droplet_size}"
+
+  private_networking = true
+  ipv6 = true
+
+  monitoring = true
+
+  ssh_keys = "${var.ssh_keys}"
+
+  connection {
+    user = "root"
+    type = "ssh"
+    private_key = "${file(var.pvt_key)}"
+    timeout = "2m"
+  }
+
+  tags = ["${var.tags}"]
+
+  volume_ids = ["${digitalocean_volume.storage.id}"]
+
+  user_data = <<BASH
+#!/bin/bash
+sudo apt-get update
```

Comment:
You should `set -euo pipefail` here, so that if the script starts to fail it dies instead of flailing and causing destruction.

Reply:
ohright